sim/core/character.go

```diff
@@ -229,21 +241,61 @@ func (character *Character) Advance(sim *Simulation, elapsedTime time.Duration)
 
 // Returns rate of mana regen, as mana / second
 func (character *Character) manaRegenPerSecond() float64 {
-	return character.stats[stats.MP5] / 5.0
+	regen := character.stats[stats.MP5] / 5.0
+	if character.PsuedoStats.SpiritRegenRateCasting > 0 {
```

Comment:
Putting any regen logic in advance() is really confusing since we're also invoking mana regen from the agent.Advance() function. This needs to get merged into the rest of the spirit regen logic in FullRegen somehow.

Reply:
the reason this is here, when calculating regen from inside of the 5 sec window you include the 'spirit regen rate casting' but in the full regen function you have to toggle between that partial spirit regen to full regen. 

 I agree, we could probably make a helper function to do this ... but I think we need to do a lot more to clean up the entire mana regen code. 

Just trying to get something basic and functional here right now.